pipeline/vars/lib.groovy

```diff
@@ -458,4 +458,71 @@ def uploadCompose(def rhBuild, def cephVersion, def baseUrl) {
     }
 }
 
+def uploadBuildRecipe(def recipeMap){
+    /*
+        This method is to PUSH Red Hat Ceph build recipe into
+        rhceph-qe-recipe github repository.
+
+        Args:
+            recipeMap   RHceph build recipe details in map
+    */
+    def recipeDir = "recipe-directory"
+    def hostAliasName = "github.com-rhceph-qe-recipe"
+    def recipeRepoUrl = "git@${hostAliasName}:red-hat-storage/rhceph-qe-recipe.git"
+    def keyPath = "~/.ssh"
+    def keyFilename = "${keyPath}/rhceph-qe-recipe-key"
+    def sshConfig = "${keyPath}/config"
+    def gitUser = "cephci"
+    def gitEmail = "cephci@redhat.com"
+
+    withCredentials([
+        sshUserPrivateKey(
+            credentialsId: 'rhceph-qe-recipe-key',
+            keyFileVariable: 'KEYFILE',
+            )
+        ]) {
+        sh "sudo cp ${KEYFILE} ${keyFilename}"
+        sh "echo -e '\nHost ${hostAliasName}\n\tHostname github.com\n\tIdentityFile ${keyFilename}' > ${sshConfig}"
+        sh "cat ${sshConfig}"
+        sh "rm -rf ${recipeDir}"
+
+        checkout([
+            $class: 'GitSCM',
+            branches: [[
+                name: '*/master'
+            ]],
+            extensions: [
+                [$class: 'RelativeTargetDirectory', relativeTargetDir: recipeDir],
+                [$class: 'CleanBeforeCheckout'],
+            ],
+            userRemoteConfigs: [
+                [
+                    credentialsId: 'rhceph-qe-recipe-key',
+                    url: recipeRepoUrl
+                ]
+            ]
+        ])
+    }
+
+    dir(recipeDir){
+        sh "git checkout master"
+        sh "git pull --rebase origin master"
+
+        def dateTime = sh (returnStdout: true, script: "date '+%D %Z %H-%M-%S'").trim()
+        recipeMap["update"] = dateTime
```

Comment:
One of the thing was,
- If content is same(This happens when you have same ceph version), we need to use force push. To avoid this introduced this `updated`, so we don't need to change git workflow used here. 

Maybe we can use `compose-id` or we can have this key-pair.

Oh, it should be `updated`.

Reply:
Okay